File: trunk/java/picocontainer/src/test/org/picocontainer/testmodel/DecoratedTouchable.java
package org.picocontainer.testmodel;

/**
 * @author Thomas Heller
 * @version $Revision$
 */
public class DecoratedTouchable implements Touchable {
    private final Touchable delegate;

    public DecoratedTouchable(Touchable delegate) {
        this.delegate = delegate;
    }

    public void touch() {
        delegate.touch();
    }
}

File: trunk/dotnet/picocontainer/src/Test/TestModel/DecoratedTouchable.cs

namespace PicoContainer.Tests.TestModel
{
	/// <summary>
	/// Summary description for DecoratedTouchable.
	/// </summary>
	public class DecoratedTouchable : Touchable
	{
		private readonly Touchable theDelegate;

		public DecoratedTouchable(Touchable theDelegate)
		{
			this.theDelegate = theDelegate;
		}

		public void touch()
		{
			theDelegate.touch();
		}

	}
}